### ISSUE TITLE
Klipper can not communicate.

### ISSUE THREAD

**Issue by ghost** [FLYmaker/FLYF407ZG#8]
When i upload klipper FW, then can not communicate klipper with MCU.
![kép](https://user-images.githubusercontent.com/48370718/73308826-625f7b00-4221-11ea-835c-e08c908f2432.png)
Maybe bootloader problem?

**Comment by FLYmaker** [comment]
I study how to use

**Comment by AniInABox** [comment]
I have the same Issue.
Kevin from Klipper said that this should be analyzed with a Hardware Debugger und seams to be a Bootloader Problem.

**Comment by kakou-fr** [comment]
waiting for it too

**Comment by kakou-fr** [comment]
Could you publish the source of the bootloader ?

**Comment by robustini** [comment]
Can someone help me?
See here:  https://github.com/FLYmaker/FLYF407ZG/issues/9

**Comment by ghost** [comment]
> Could you publish the source of the bootloader ?

I need this bootloader too.

**Comment by AniInABox** [comment]
Does something happen in the Background, or can i put the Board into trash?

Please give US the Bootloader Code vor the Wiring Diagramm. Wrn want to usw this Board with Klipper and other Firmwares.

**Comment by ghost** [comment]
Work with Marlin!

AniInABox <notifications@github.com> ezt írta (időpont: 2020. febr. 16., V
16:28):

**Comment by ghost** [comment]
My opinion is: FLYmaker user is chinese. New coronavirus is much more important than FLYboard.

**Comment by FLYmaker** [comment]
klipper FW requires DVECT_TAB_OFFSET = 0x8000
  The bootloader writes the klipper FW to the address after 0x8008000

**Comment by FLYmaker** [comment]
> 我的意见是：FLYmaker用户是中国人。新的冠状病毒比FLYboard重要得多。

thank you
Thankfully i did not get sick

**Comment by AniInABox** [comment]
> 
> 
> klipper FW requires DVECT_TAB_OFFSET = 0x8000
>   The bootloader writes the klipper FW to the address after 0x8008000

What does that mean? I have to configure Klipper to Write to another Start Position?

**Comment by AniInABox** [comment]
Here is the Answer from Kevin (Klipper):

Setting a "32KiB bootloader" in "make menuconfig" sets the firmware address to 0x8008000. So, the problem must be something else.

-Kevin

**Comment by kakou-fr** [comment]
just tested, it work with klipper

**Comment by SmoothieAq** [comment]
@kakou-fr could you please give some details on how you made it work.

I tried:

1. make menuconfig with:   
   - Micro-controller Architecture: STMicroelctronics STM32
   - Processor model: STM32F407
   - Bootloader offset: 32KiB
   - Clock reference: 8 MHz crystal
   - Use USB instead of serial
2. make the klipper.bin
3. format a 8GB sd card on my Windows pc with FAT32
4. Copy the klipper.bin to that
5. Place the sd card in the FLY and connect it with USB to the Raspberry
6. Power off, then power on

When I look in /dev, I can not see anything that looks like an usb port the the FLY
On the FLY I have 3 green leds

I am at loss what to do now, help appreciated :-)

**Comment by kakou-fr** [comment]
│ │                           [ ] Enable extra low-level configuration options                                                 │ │
  │ │                               Micro-controller Architecture (STMicroelectronics STM32)  --->                               │ │
  │ │                               Processor model (STM32F407)  --->                                                            │ │
  │ │                               Bootloader offset (32KiB bootloader (SKR-PRO))  --->                                         │ │
  │ │                           [*] Use USB for communication (instead of serial)                                                │ │
  │ │                               USB ids  --->


copy klipper/bin and rename it firmware.bin

**Comment by SmoothieAq** [comment]
yeah, forgot that. Now I get an usb-Klipper_xxx in /dev/serial/by-id :-D
Thank you!

I'll try to configure the first parts of klipper this weekend...

**Comment by SmoothieAq** [comment]
It is up an running now.
Seems you just have to read the 10 line instruction in generic-flyboard.cfg and do what it says. What with all the negative fuzz??
